vshard/service_info.lua

```diff
@@ -45,6 +46,15 @@ local function service_set_status_ok(service)
     service.error = ''
 end
 
+local function service_log_error_if_needed(service, level, err, ...)
+    local err_str = string.format(err, ...)
+    if not service.is_error_set then
+        level = service.error ~= err_str and level or 'verbose'
+        log[level](err_str)
+    end
+    return err_str
```

Comment:
It is quite counter-intuitive that the error message is returned. Lets better just take it as a ready message and return nothing. Otherwise, if a caller would just want to log an error message received from somewhere else, then it would have to be escaped from any special characters to avoid `string.format()` doing anything crazy with it.

Or merge it with `set_status_error()` into one function.